test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,348 @@
-        tensor, mask = nt.to_tensor_mask()
-        self.assertTrue([len(value) for value in values] == list(mask.sum(-1)))
+    def test_gen_nested_tensor(self):
+        nt1 = utils.gen_nested_tensor(seed=1, nested_dim=1, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(nt.is_nested_tensor(nt1))
+        self.assertEqual(nt1.nested_dim(), 1)
+        self.assertEqual(nt1.size(), (2, 2))
+        self.assertEqual(nested_size_to_list(nt1.nested_size()), [[2],[2]])
+        self.assertEqual(nt1[0].dim(), 1)
+        self.assertEqual(nt1[0].size(), torch.Size([2]))
+
+        nt1 = utils.gen_nested_tensor(seed=1, nested_dim=2, tensor_dim=1, size_low=2, size_high=2)
+        self.assertTrue(nt.is_nested_tensor(nt1))
+        self.assertEqual(nt1.nested_dim(), 2)
+        self.assertEqual(nt1.size(), (2, 2, 2))
+        self.assertEqual(nt1[0].dim(), 2)
+        self.assertEqual(nested_size_to_list(nt1.nested_size()), [[[2],[2]], [[2],[2]]])
+
+        nt1 = utils.gen_nested_tensor(seed=3, nested_dim=2, tensor_dim=2, size_low=1, size_high=5)
+        self.assertTrue(nt.is_nested_tensor(nt1))
+        self.assertEqual(nt1.nested_dim(), 2)
+        self.assertEqual(nt1.size(), (3, 2, None, None))
+        self.assertEqual(nt1[0].dim(), 3)
+        self.assertEqual(nt1[0].size(), (2, None, None))
+        self.assertEqual(nt1[0][0].size(), torch.Size([2, 4]))
+        self.assertEqual(nt1[0][0][0].size(), torch.Size([4]))
+        self.assertEqual(nested_size_to_list(nt1.nested_size()), [[[2, 4], [4, 3]], [[2, 4], [4, 3]], [[2, 4], [4, 3]]])
+
+    #
+    # Group of tests to test to_tensor_mask() 
+    # 
+    def test_to_tensor_mask_nt_dim_2(self):
+        nt1 = nt.nested_tensor([
+            torch.tensor([1, 2]),
+            torch.tensor([3, 4])
+        ])
+        exp_tensor = torch.tensor([[1, 2], [3, 4]])
+        exp_mask = torch.tensor(True)
+
+        res_tensor, res_mask = nt1.to_tensor_mask()
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+
+    def test_to_tensor_mask_nt_dim_4(self):
+        original_nt2 = nt.nested_tensor([
+            nt.nested_tensor([
+                nt.nested_tensor([
+                    torch.tensor([1, 2])
+                ]),
+            ]),
+            nt.nested_tensor([
+                nt.nested_tensor([
+                    torch.tensor([3, 4]),
+                    torch.tensor([5, 6])
+                ])
+            ])
+        ])
+
+        # mask_dim = None
+        res_tensor, res_mask = original_nt2.to_tensor_mask()
+        exp_mask = torch.tensor([[[ True, False]],
+                                 [[ True, True]]])
+        exp_tensor = torch.tensor([[[[1, 2], [0, 0]]],
+                                   [[[3, 4], [5, 6]]]])
+
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+
+        # mask_dim = 0
+        res_tensor, res_mask = original_nt2.to_tensor_mask(mask_dim=0)
```

Comment:
There's a choice we can make here. We can raise an error since it's not constructible with that mask dimension or do what you did. I'd prefer an error (I know that changing the code you're testing isn't part of testing) because it increases identifiability.